kedro-datasets/kedro_datasets_experimental/pandas/delta_sharing_dataset.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+from typing import Any
+
+import delta_sharing
+import pandas as pd
+from kedro.io.core import AbstractDataset, DatasetError
+
+from kedro_datasets._typing import TablePreview
+
+
+class DeltaSharingDataset(AbstractDataset):
+    """``DeltaSharingDataset`` loads data from a Delta Sharing shared table using the Delta Sharing open protocol.
+    This dataset handles loading data into a Pandas DataFrame. Saving to Delta Sharing is not supported.
+
+    Delta Sharing is an open protocol for secure real-time exchange of large datasets, which enables
+    organizations to share data in real time regardless of which computing platforms they use. It is a
+    simple REST protocol that securely shares access to part of a cloud dataset and leverages modern cloud
+    storage systems, such as S3, ADLS, or GCS, to reliably transfer data.
+
+    Example usage for the YAML API:
+
+    .. code-block:: yaml
+
+        my_delta_sharing_dataset:
+          type: pandas.DeltaSharingDataset
+          share: <share-name>
+          schema: <schema-name>
+          table: <table-name>
+          credentials:
+            profile_file: <profile-file-path>
+          load_args:
+            version: <version>
+            limit: <limit>
+            use_delta_format: <use_delta_format>
+
+    Example usage for the Python API:
+
+    .. code-block:: pycon
+
+        >>> from kedro_datasets_experimental.pandas import DeltaSharingDataset
+        >>>
+        >>> credentials = {"profile_file": "conf/local/config.share"}
+        >>> load_args = {"version": 1, "limit": 10, "use_delta_format": True}
+        >>> dataset = DeltaSharingDataset(
+        ...     share="example_share",
+        ...     schema="example_schema",
+        ...     table="example_table",
+        ...     credentials=credentials,
+        ...     load_args=load_args,
+        ... )
+        >>> data = dataset.load()
+        >>> print(data)
```

Comment:
```suggestion
    Example usage for the Python API:
    
    To start quickly, you can try this dataset with the [hosted service](https://github.com/delta-io/delta-sharing?tab=readme-ov-file#accessing-shared-data) for sample dataset. You will need to download the credentials with this [link](https://databricks-datasets-oregon.s3-us-west-2.amazonaws.com/delta-sharing/share/open-datasets.share).

    .. code-block:: pycon

        >>> from kedro_datasets_experimental.pandas import DeltaSharingDataset
        >>>
        >>> credentials = {"profile_file": "conf/local/config.share"}
        >>> load_args = {"version": 1, "limit": 10, "use_delta_format": True}
        >>> dataset = DeltaSharingDataset(
        ...     share="example_share",
        ...     schema="example_schema",
        ...     table="example_table",
        ...     credentials=credentials,
        ...     load_args=load_args,
        ... )
        >>> data = dataset.load()
        >>> print(data)
```

Reply:
Or maybe

```
kedro_datasets_experimental.delta_sharing import PandasDeltaSharingDataset, SparkDeltaSharingDataset
```

?

Reply:
I would actually expect this to be just a type for a generic delta sharing dataset.

Reply:
> from pyspark.sql import SparkSession
> 
> spark = SparkSession.builder.config(
>     "spark.jars.packages", "io.delta:delta-sharing-spark_2.12:3.1.0"
> ).getOrCreate()

I try to setup the spark with delta-sharing support, and it seems to work quite well by just changing one line in the `_load` method with `load_as_spark` instead of `load_as_pandas`.